### PR TITLE
Authorization

### PR DESCRIPTION
**Pull request by blangley28**
Implement SMART Backend Authorization Protocol. Requests to the backend app are now protected (using passport middleware) and require an access token from Keycloak. For testing purposes "admin" is also hard coded as a valid token like it is in the FHIR servers.

### CODE REVIEW

File: src/utils/auth.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * Helper function to obtain the token from the authorization header
+ *
+ * @param {*} req - the express request
+ * @returns the bearer token if it exists, otherwise null
+ */
+function getToken(req) {
+  const authHeader = req.headers.authorization;
+  if (authHeader) return authHeader.split(' ')[1];
```
Comment (dehall) [inline src/utils/auth.js:11]: Might be good just as a reminder to comment that this expects the header to be of the form "Bearer {token}".

File: src/app.js [hunk 2]
```diff
@@ -3,21 +3,43 @@ require('dotenv').config();
...
 app.use(logger('dev'));
 app.use(express.json());
 app.use(bodyParser.json({ type: ['application/json', 'application/fhir+json'] }));
+app.use(passport.initialize());
+
+passport.use(
+  new KeycloakBearerStrategy(
+    {
+      realm: 'backend_app',
+      url: 'http://moonshot-dev.mitre.org:8090/auth'
```
Comment (dehall) [inline src/app.js:28]: Can this url and the token endpoint in `src/routes/wellknown.js` be made configurable? Those would work well in a .env file. Maybe we should also have a .env.dev or .env.mitre or something with these pre-populated.

Comment (blangley28) [reply src/app.js]: Added REALM and AUTH env vars which are required. Pushed .env.dev which contains an example .env file

**Review by dehall — APPROVED** [review]

**Comment by KeeyanGhoreshi** [comment]
is there a way to test this without using the test `admin` token?

**Review by KeeyanGhoreshi — APPROVED** [review]
creating a JWT and posting to the auth server returns an auth token that the backend server accepts.  Works and the code looks fine.

**Pull request merged**